File: ios/checkForgery.swift
import Foundation
import SwiftKeychainWrapper
import Alamofire
import SwiftyJSON

@objc(checkForgery)
public class checkForgery : NSObject {
  var session : Session? = nil
  @objc
  func isForgery(_ successCallback: @escaping RCTResponseSenderBlock, errorCallback: @escaping RCTResponseSenderBlock) {
    
    var version: String? {
      guard let dictionary = Bundle.main.infoDictionary,
          let version = dictionary["CFBundleShortVersionString"] as? String else {return nil}
      
      return "\(version)"
    }
//    let urlString = "https://ompass.kr:8800/oms/app-verification/os/ios/version/" + version!
    let urlString = "https://admin-api.ompasscloud.com/oms/app-verification/os/ios/version/" + version!
    let configuration = URLSessionConfiguration.af.default
    configuration.timeoutIntervalForRequest = 5
//    let manager = ServerTrustManager(allHostsMustBeEvaluated: false, evaluators: ["https://ompass.kr:8800": DisabledTrustEvaluator()])
    let manager = ServerTrustManager(allHostsMustBeEvaluated: false, evaluators: ["https://admin-api.ompasscloud.com": DisabledTrustEvaluator()])
    session = Session(configuration: configuration, serverTrustManager: manager)
    let url = NSURL(string: urlString)
    var request = URLRequest(url: url! as URL)
    request.httpMethod = "GET"
    request.setValue("application/json", forHTTPHeaderField: "Content-Type")
    request.setValue("application/json", forHTTPHeaderField: "accept")
    session?.request(request)
      .responseJSON { response in
        
        switch response.result {
        case .success(let value):
          if let data = response.data {
            do {
              let json = try JSON(data: data)
              let responseJson = JSON(json["data"])
              let result: [String: Any] = [
                "version" : responseJson["version"].boolValue,
                "hash": true
              ] as Dictionary
              successCallback([result])
            }
            catch {
              errorCallback(["Internel Error"])
            }
          }
        case.failure(let error):
          errorCallback(["Server Connection Error"])
        }
      }
  }
}

File: android/app/src/main/java/kr/omsecurity/ompass/CheckForgery.java
package kr.omsecurity.ompass;

import android.annotation.SuppressLint;
import android.content.pm.PackageManager;
import android.os.Build;
import android.util.Log;

import androidx.annotation.NonNull;
import com.facebook.react.bridge.*;
import kr.omsecurity.ompass.Constants.StaticMethods;
import org.json.JSONException;
import org.json.JSONObject;

import javax.net.ssl.HttpsURLConnection;
import javax.net.ssl.SSLHandshakeException;
import java.io.*;
import java.net.ConnectException;
import java.net.MalformedURLException;
import java.net.URL;
import java.nio.charset.StandardCharsets;
import java.security.*;
import java.security.cert.CertificateException;
import java.util.Iterator;

import static kr.omsecurity.ompass.webauthn.AuthenticatorManager.insertCAContext;

public class CheckForgery extends ReactContextBaseJavaModule {
    private static ReactApplicationContext reactContext;

    @SuppressLint("CommitPrefEdits")
    public CheckForgery(ReactApplicationContext context) {
        super(context);
        reactContext = context;
    }

    @Override
    @NonNull
    public String getName() {
        return "checkForgery";
    }

    @ReactMethod
    public void isForgery(Callback successCallback, Callback errorCallback) {
        String err_msg = null;
        try {
            String installer = null;
            PackageManager pm = reactContext.getPackageManager();
            String packageName = reactContext.getPackageName();
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
                try {
                    installer = pm.getInstallSourceInfo(packageName).getInstallingPackageName();
                } catch (PackageManager.NameNotFoundException e) {
                    e.printStackTrace();
                }
            } else {
                installer = pm.getInstallerPackageName(packageName);
            }
            if (BuildConfig.DEBUG || (installer != null && installer.startsWith("com.android.vending"))) {
//            if (true) {
                URL urlAuthenticate = new URL("https://admin-api.ompasscloud.com/oms/app-verification/os/android/version/" + BuildConfig.VERSION_NAME);
                HttpsURLConnection conCheckForgery = null;
                conCheckForgery = (HttpsURLConnection) urlAuthenticate.openConnection();
                conCheckForgery.setSSLSocketFactory(insertCAContext(R.raw.thawte_rsa_ca_2018, reactContext).getSocketFactory());
                conCheckForgery.setRequestMethod("GET");
                conCheckForgery.setRequestProperty("Content-Type", "application/json; utf-8");
                conCheckForgery.setRequestProperty("Accept", "application/json");
                conCheckForgery.setConnectTimeout(5000);

                try (BufferedReader br = new BufferedReader(
                        new InputStreamReader(conCheckForgery.getInputStream(), StandardCharsets.UTF_8))) {
                    StringBuilder response = new StringBuilder();
                    String responseLine = null;
                    while ((responseLine = br.readLine()) != null) {
                        response.append(responseLine.trim());
                    }
                    JSONObject json = new JSONObject(response.toString());
                    JSONObject jsonData = json.getJSONObject("data");
                    jsonData.put("hash", true);
                    successCallback.invoke(StaticMethods.convertJsonToMap(jsonData));
                } catch (FileNotFoundException | SSLHandshakeException | ConnectException e) {
                    e.printStackTrace();
                    err_msg = "Server Connection Error";
                }
            } else {
                WritableMap result = new WritableNativeMap();
                result.putBoolean("hash", false);
                successCallback.invoke(result);
            }
        } catch (IOException | NoSuchAlgorithmException | CertificateException | KeyStoreException |
                 KeyManagementException | JSONException e) {
            e.printStackTrace();
            err_msg = "Internel Error";
        }
        if (err_msg != null) errorCallback.invoke(err_msg);
    }
}